src/common/AndroidUtils.ts

```diff
@@ -613,23 +570,133 @@ export class AndroidSDKUtils {
         });
     }
 
+    public static getAndroidPlatformTools(): string {
+        if (!AndroidSDKUtils.androidPlatformTools) {
+            AndroidSDKUtils.androidPlatformTools = path.join(
+                AndroidSDKUtils.getAndroidSdkRoot(),
+                'platform-tools'
+            );
+        }
+
+        return AndroidSDKUtils.androidPlatformTools;
+    }
+
+    public static getAndroidSdkRoot(): string {
+        if (!AndroidSDKUtils.sdkRoot) {
+            const home =
+                process.env.ANDROID_HOME && process.env.ANDROID_HOME.trim();
+
+            const root =
+                process.env.ANDROID_SDK_ROOT &&
+                process.env.ANDROID_SDK_ROOT.trim();
+
+            if (home && fs.existsSync(home)) {
+                AndroidSDKUtils.logger.info(
```

Comment:
I know this is going to require some flow-through interface changes, but I think we need to surface this output more visibly to the user, at the `android:reqs:androidhome:fulfilledMessage` level. I think giving such output will head off potential configuration issues for the end user.

Reply:
Yes we can definitely do that. There are a few ways of doing it, one of which is to update `getAndroidSdkRoot()` to return a tuple which includes the path and the source where we got the path. Another is to introduce a new method `getAndroidSdkRootWithSource()` in addition to `getAndroidSdkRoot()`. Do you have a preference? Or should we do something else entirely?

Reply:
I was thinking the tuple route too (or some encapsulating object that effectively does that), and just update the existing signature (and consumers) to return/use those values. Works for me if it works for you.